lib/child-writer/index.js

```diff
@@ -113,25 +113,36 @@ exports.write = (image, drive, options) => {
     emitter.emit('error', error);
   };
 
-  ipc.server.on('error', emitError);
-  ipc.server.on('message', (data) => {
-    let message = null;
-
+  /**
+   * @summary Bridge robot message to the child writer caller
+   * @function
+   * @private
+   *
+   * @param {String} message - robot message
+   *
+   * @example
+   * bridgeRobotMessage(robot.buildErrorMessage(new Error('foo')));
+   */
+  const bridgeRobotMessage = (message) => {
     try {
-      message = robot.parseMessage(data);
-    } catch (error) {
-      return emitError(error);
-    }
+      const parsedMessage = robot.parseMessage(message);
+
+      // The error object is decomposed by the CLI for serialisation
+      // purposes. We compose it back to an `Error` here in order
+      // to provide better encapsulation.
+      if (robot.getCommand(parsedMessage) === 'error') {
+        emitError(robot.recomposeErrorMessage(parsedMessage));
+      } else {
+        emitter.emit(robot.getCommand(parsedMessage), robot.getData(parsedMessage));
```

Comment:
Could we have a single function that returned both the `command` and `data` (e.g. as a tuple) at the same time? Or maybe to keep it more generic, it should be a function which accepts a list of properties, and returns an object with those properties set?

Or maybe `parsedMessage` itself should be an Object with accessor functions (this is where my JS knowledge runs out) so you could do `parsedMessage.getCommand()` and `parsedMessage.getData()` ?

If none of those make sense, or if they'd make the code less readable, then feel free to keep things as-is.

Alternatively, how likely is it that `robot.getCommand(parsedMessage)` will ever do anything _other_ than returning `parsedMessage.command` ?

\*shrug\*  You're obviously for more familiar with usual JS practices then I am.